classes/classes/Scenes/Areas/GlacialRift/Amarok.as

```diff
@@ -0,0 +1,164 @@
+			}
+			else if (player.getEvasionRoll()) {
+				outputText("The Amarok throws itself at you, attempting to slash you with its claws. Luckily, you manage to leap out of the way.", false);
+				combatRoundOver();
+				return;
+			}
+			else {
+				//Damage roll
+				outputText("The Amarok throws itself at you and rakes one of its hefty paws across you. Its claws slice you open and draw blood.", false);
+				var damage:int = ((str + 50) + rand(100));
+				damage = player.reduceDamage(damage);
+				player.takeDamage(damage, true);
+				if(player.HP <= 0) {
+					doNext(game.combat.endHpLoss);
+					return;
+				}
+			}
+			combatRoundOver();
+		}
+		protected function amarokTail():void {
+			//AMAROK used TAIL SLAP!
+			outputText("The Amarok rushes up to you and immediately turns heel, attempting to crash its tail into you. ", false);
+			//Blind check...
+			if(findStatusEffect(StatusEffects.Blind) >= 0) {
+				outputText("Luckily, though, its blindness causes it to misjudge your location and it misses entirely.", false);
+				combatRoundOver();
+				return;
+			}
+			//Evasion stuffs
+			if(player.findPerk(PerkLib.Evade) >= 0 && player.spe >= 35 && rand(3) != 0) {
+				outputText("You manage easily jump out of the way due to your talents in evasion.", false);
+			}
+			else if(player.findPerk(PerkLib.Misdirection) >= 0 && rand(100) < 20 && player.armorName == "red, high-society bodysuit") {
+				outputText("However, Raphael's teachings and your bodysuit's flexibility let you get out of the way before it can connect.", false);
+			}
+			else if(player.findPerk(PerkLib.Flexibility) >= 0 && player.spe > 30 && rand(10) != 0) {
+				outputText("You use your incredibly flexibility and twist out of the way of the attack.", false);
+			}
+			else if (player.getEvasionRoll()) {
+				outputText("You jump out of the way before it can hit.", false);
+				combatRoundOver();
+				return;
+			}
+			else {
+				//Damageeee + stun! Reference to the legend of it slapping a kid with its tail, except minus the bone breaking.
+				outputText("The hit sends you stumbling back", false);
+				if (player.findPerk(PerkLib.Resolute) < 0 && rand(2) == 0) 
+				{
+					outputText(", stunning you.", false);
+					player.createStatusEffect(StatusEffects.Stunned, 0, 0, 0, 0);
+				}
+				else outputText(".", false);
+				var damage:int = ((str + 100) + rand(75));
+				damage = player.reduceDamage(damage);
+				player.takeDamage(damage, true);
+				if(player.HP <= 0) {
+					doNext(game.combat.endHpLoss);
+					return;
+				}
+			}
+			combatRoundOver();
+		}
+		
+		override public function defeated(hpVictory:Boolean):void
+		{
+			game.glacialRift.amarokScene.winAgainstAmarok();
+		}
+		
+		override public function won(hpVictory:Boolean, pcCameWorms:Boolean):void
+		{
+			if(pcCameWorms){
+				outputText("\n\nThe Amarok looks down at the worms you came, sniffs them, then snarls and walks away. It must consider you tainted meat.", false);
```

Comment:
It's a reference to the fact that in the bad end, the Amarok literally eats you. It sees you as prey. So, if you're infested with worms, you are tainted prey (hence to "meat"). It's meant to be unsettling text. You're literally being hunted.